File: core/java/src/main/java/hara/lang/protocol/IIndexed.java
package hara.lang.protocol;

import hara.lang.declaration.HaraMethod;
import hara.lang.declaration.HaraProtocolBinding;

@HaraProtocolBinding(namespace = "std.protocol.iindexed", name = "IIndexed")
public interface IIndexed<K, V> {
  @HaraMethod(value = "index-of", arity = 2)
  K indexOf(V val);
}

File: core/rust/src/lang/protocol/iindexed.rs
use hara_protocol_macros::hara_protocol;

#[hara_protocol(namespace = "std.protocol.iindexed", name = "IIndexed")]
pub trait IIndexed<V> {
    type Index;

    #[hara_method(value = "index-of", arity = 2)]
    fn index_of(&self, value: &V) -> Option<Self::Index>;
}
